x-pack/plugins/endpoint/server/routes/resolver.ts

```diff
@@ -0,0 +1,130 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { IRouter } from 'kibana/server';
+import { SearchResponse } from 'elasticsearch';
+import { schema } from '@kbn/config-schema';
+import { EndpointAppContext, Total } from '../types';
+import {
+  getESChildrenQuery,
+  getESNodeQuery,
+  getESChildrenCountQuery,
+  getESNodeCountQuery,
+  PaginationInfo,
+} from '../services/resolver/query_builder';
+import { ResolverSearchHandler } from '../services/resolver/search_handler';
+import { ResolverData } from '../../common/types';
+import { EntityParseError } from '../services/resolver/common';
+
+export function registerResolverRoutes(router: IRouter, endpointAppContext: EndpointAppContext) {
+  const log = endpointAppContext.logFactory.get('resolver-route');
+  const validateQueryObject = schema.object({
+    page_size: schema.maybe(schema.number({ min: 1, max: 1000 })),
```

Comment:
Is `schema.maybe` necessary here?

Reply:
I believe the query params is required if we don't use `schema.maybe`. I was just making it a convenience thing for the caller. If they aren't included than we will default it to the value defined in `server/configs`. We could put the default in the schema call here but then we can't change them with a configuration file 🤷‍♂